tests/integration_test/isomorphism_problem.py

```diff
@@ -1,29 +1,45 @@
 from algorithms.preprocessing import fix_degrees
 from tests.integration_test.algorithm_options import apply_could_be_isomorphic, apply_remove_twins, apply_tree_algorithm, branching_method, apply_complement
+from algorithms.color_initialization import degree_color_initialization
+
+
+def preprocessing(G: "Graph"):
+    """
+    In this method all pre-processing is done that need to be done only one time. This method is called before any
+    other computations are done.
+    :param G: The graph to be preprocessed
+    :return: Factor the amount of automorphisms need to be multiplied with to get the right result
+    """
+    G_complement = apply_complement(G)
+
+    fix_degrees(G)
+    fix_degrees(G_complement)
+
+    # Twin removal
+    factor = apply_remove_twins(G)
+
+    return G_complement, factor
+
 
-"""
-With these methods, the graph isomorphism problem can be solved.
-"""
 def are_isomorph(G: "Graph", H: "Graph"):
     """
     This method determines if graph G and graph H have at least one isomorphism.
     :param G, H: The two graphs of which it must be determined if there is an isomorphism.
     :return: Boolean that indicates if graph G and H are isomorph or not.
     """
-    fix_degrees(G), fix_degrees(H)
-
+    # Test if graphs are isomorphic using simple properties of the graphs
     if not apply_could_be_isomorphic(G, H):
         return False
 
-    G = apply_complement(G)
-    H = apply_complement(H)
-
-    G_disjoint_union, _ = apply_remove_twins(G, H)
-
+    # If graph is a tree, use this algorithm to solve the GI problem
     problem_solved, is_isomorph = apply_tree_algorithm(G, H, False)
```

Comment:
Is hiervoor geen kleuring nodig?

Reply:
Nee het tree algoritme gebruikt geen kleuring